lib/Db/QueueMapper.php

```diff
@@ -82,6 +83,27 @@ public function removeFileFromAllQueues(int $fileId) : void {
 		}
 	}
 
+	/**
+	 * @param string $model
+	 * @param \OCA\Recognize\Db\QueueFile $file
+	 * @return \OCA\Recognize\Db\QueueFile|null
+	 * @throws \OCP\AppFramework\Db\MultipleObjectsReturnedException
+	 * @throws \OCP\DB\Exception
+	 */
+	public function existsQueueItem(string $model, QueueFile $file) : ?QueueFile {
+		$qb = $this->db->getQueryBuilder();
+		$qb->select(QueueFile::$columns)
+			->from($this->getQueueTable($model))
+			->where($qb->expr()->eq('file_id', $qb->createPositionalParameter($file->getFileId(), IQueryBuilder::PARAM_INT)))
+			->setMaxResults(1);
+
+		try {
+			return $this->findEntity($qb);
+		} catch (DoesNotExistException $e) {
```

Comment:
If it returns a bool you could also catch the MultipleObjectsReturnedException here and simplify the try/catch in https://github.com/nextcloud/recognize/pull/478/files#diff-015c8d4799471110c51625847e6dbafe4a476fb46461d1eca03cd9e35132a234R51-R58 a bit